### ISSUE TITLE
Jersey's internal validation logic not works properly with @ValidateOnExecution set to ExecutableType.NONE

### ISSUE THREAD

**Issue by dmaidaniuk** [jersey/jersey#3582]
[jersey-reproducer.txt](https://github.com/jersey/jersey/files/1072566/jersey-reproducer.txt)
It looks like in Jersey exists a bug with processing `@ValidateOnExecution` annotation properly.
I prepared a small reproducer project where in `FormController` we have next code snippet:
```
@Path("form")
public class FormController {

    @POST
    @Produces(MediaType.TEXT_PLAIN)
    @ValidateOnExecution(type = ExecutableType.NONE)
    public Response formPost(@Valid @BeanParam FormDataBean form) {
        return Response.status(OK).entity(form.toString()).build();
    }

}
```
Thus according to [Bean Validation 1.1 documentation](http://beanvalidation.org/1.1/spec/#integration-general-executable) given method shouldn't be validated by Jersey's internal validation mechanism. Same statement present also in [Jersey's own documentation](https://jersey.github.io/documentation/latest/user-guide.html#bv.ValidateOnExecution):

> The special annotation @ValidateOnExecution can be used to selectively enable and **disable** validation. 

However annotation `@ValidateOnExecution(type = ExecutableType.NONE)` ignored by Jersey and HTTP code 400 (Bad Request) returned to client in case with not valid data send in form, which is wrong.

I attached reproducer project to this issue report (on top of message). Just change file extension from _.txt_ to _.zip_ and unzip it.

**Comment by chkal** [comment]
I just reproduced this with the latest promoted build of Glassfish 5.

**Comment by pavelbucek** [comment]
As creative as adding zipped file renamed to txt might be, creating a github repo and linking it from the issue would be more helpful.

Thanks for the report, seems like Jersey is not considering @ValidateOnExecution on resource methods in a way it should. There is also related thread on the JAX-RS EG: https://javaee.groups.io/g/jaxrs-spec/topic/jax_rs_and/5223794?p=,,,100,0,0,0::recentpostdate%2Fsticky,,,100,2,0,5223794

**Comment by pavelbucek** [comment]
just for the reference: https://github.com/pavelbucek/jersey/commit/01ef7993849f1779b5c6f7d2fe8fa25a7f4b546b

**Comment by dmaidaniuk** [comment]
Reproducer repository: https://github.com/dmaidaniuk/jersey-3582-reproducer

**Comment by chkal** [comment]
@pavelbucek Awesome! Many thanks for addressing this issue!

**Comment by pavelbucek** [comment]
@chkal don't thank yet, the issue is not fixed.

The change caused multiple regressions, which need to be carefully evaluated..

**Comment by dmaidaniuk** [comment]
Hi @pavelbucek,

Thanks for fix! Do you know in which version Jersey it will be included?

Regards,
Dmytro

**Comment by chkal** [comment]
This fix has been released in Jersey 2.26-b07. 

The version of Jersey in Glassfish 5 has been updated a week ago. So the new Jersey version is part of the latest Glassfish nightly builds, but not yet in the promoted builds.
